server/athenian/api/internal/miners/github/deployment.py

```diff
@@ -2022,3 +2024,132 @@ async def load_jira_issues_for_deployments(deployments: pd.DataFrame,
                 prs[pri] = np.sort(np.array(release_keys, dtype="S"))
 
     return issues
+
+
+async def hide_anomal_first_deployments(
+    account: int,
+    meta_ids: Sequence[int],
+    deployment_facts: pd.DataFrame,
+    pdb: Database,
+    mdb: Database,
+    threshold: float = 100.0,
+) -> None:
+    """Hide the anomal first deployments by deleting their facts from pdb."""
+    log = logging.getLogger(f"{__name__}.hide_first_deployments")
+    anomal_deploys = await _search_anomal_first_deployments(
+        meta_ids, deployment_facts, mdb, threshold,
+    )
+
+    anomal_deploys = anomal_deploys
+    if not anomal_deploys:
+        return
+
+    log.info("Clearing %d anomal first deployments", len(anomal_deploys))
+
+    stmts = []
+
+    tables = (GitHubCommitDeployment, GitHubPullRequestDeployment, GitHubReleaseDeployment)
+    for Table in tables:
+        where_clause = sa.or_(*(
+            sa.and_(
+                Table.acc_id == account,
+                Table.repository_full_name == deploy.repository_full_name,
+                Table.deployment_name == deploy.deployment_name,
+            ) for deploy in anomal_deploys
+        ))
+        stmts.append(sa.delete(Table).where(where_clause))
+
+    # clear GitHubDeploymentFacts by removing prs and commits from facts data
+    depl_facts_stmt = sa.select(GitHubDeploymentFacts).where(
+        sa.and_(
+            GitHubDeploymentFacts.acc_id == account,
+            GitHubDeploymentFacts.deployment_name.in_(d.deployment_name for d in anomal_deploys),
+        ),
+    )
+    depl_facts_rows = await pdb.fetch_all(depl_facts_stmt)
+    for row in depl_facts_rows:
+        facts = DeploymentFacts(
+            row[GitHubDeploymentFacts.data.name],
+            name=row[GitHubDeploymentFacts.deployment_name.name],
+        )
+        new_facts = facts.with_no_prs_commits()
```

Comment:
```suggestion
        new_facts = facts.with_nothing_deployed()
```
prs, commits, releases

Reply:
done